arches/app/models/models.py

```diff
@@ -874,15 +894,18 @@ def __init__(self, *args, **kwargs):
     def clean(self):
         if not self.alias:
             Graph.objects.get(pk=self.graph_id).create_node_alias(self)
+        if self.pk == self.source_identifier_id:
+            self.source_identifier_id = None
 
     def save(self, **kwargs):
         if not self.alias:
-            self.clean()
             add_to_update_fields(kwargs, "alias")
             add_to_update_fields(kwargs, "hascustomalias")
         if self.pk == self.source_identifier_id:
-            self.source_identifier_id = None
```

Comment:
...(continuing from last comment)

This is where the logic moved from. It was already in save(), and we already had a mixture of a clean method but without other-cleaning-logic-inside-save. If we want to move to model methods only, I'm all for it, and then we can stop overriding save()!

Anyway -- I can set this back if you don't like it, it's not strictly necessary, but while I was in the area I thought a little cleaner separation between clean and save would be good. Thoughts?

Reply:
yeah no it's definitely a larger conversation but for sure out of scope for this PR 👍 . I won't block on it 😄